File: backend/controle.gastos.Domain/Entities/Transacoes.cs
using controle.gastos.Domain.Enums;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace controle.gastos.Domain.Entities;

public class Transacoes
{
    [Key]
    public int Id { get; set; }

    [Required, StringLength(400)]
    public string Descricao { get; set; } = string.Empty;

    [Required]
    public decimal Valor { get; set; }

    [Required]
    public Tipo Tipo { get; set; }

    [Required]
    public DateTime Data { get; set; }

    [Required]
    public int PessoaId { get; set; }

    [Required]
    public int CategoriaId { get; set; }



    [ForeignKey(nameof(PessoaId))]
    public Pessoas Pessoa { get; set; } = null!;

    [ForeignKey(nameof(CategoriaId))]
    public Categorias Categoria { get; set; } = null!;
}

File: frontend/types/transacoes.ts
import { Categoria } from "./categorias"
import { Pessoa } from "./pessoas"

export interface Transacao {
    id: number
    descricao: string
    valor: number
    tipo: 0|1  // 0 = Despesa, 1 = Receita
    data: string
    pessoaId: number
    categoriaId: number
    pessoa?: Pessoa
    categoria?: Categoria
}

export interface FiltroTransacao {
    Id?: number
    Descricao?: string
    ValorInicial?: number
    ValorFinal?: number
    Tipo?: 0|1
    DataInicio?: string
    DataFim?: string
    Pessoas?: string
    Categorias?: string
}

export interface NovaTransacao {
    Descricao: string
    Valor: number
    Tipo: string
    Data: string
    PessoaId: number
    CategoriaId: number
}

export interface AlteraTransacao {
    Id: number
    Descricao?: string
    Valor?: number
    Tipo?: string
    Data?: string
    PessoaId?: number
    CategoriaId?: number
}

